backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1758,3 +1758,96 @@ databaseChangeLog:
             columns:
               - column:
                   name: test_result_delivery
+  - changeSet:
+      id: add-spring-session-jdbc-support
+      author: emma@skylight.digital
+      comment: Add a table to back spring sessions.
+      changes: 
+        - createTable:
+              tableName: spring_session
+              remarks: Data store for user sessions, as created by SpringSession.
+              columns:
+                - column:
+                    name: primary_id
+                    type: *idtype
+                    remarks: The immutable internal key to identify this session.
+                    constraints:
+                      nullable: false
+                      primaryKey: true
+                      primaryKeyName: pk__spring_session
+                - column:
+                    name: session_id
+                    type: *idtype
+                    remarks: A unique, externally visible string that identifies this session.
+                    constraints:
+                      nullable: false
+                      unique: true
+                      uniqueConstraintName: uk__spring_session
+                - column:
+                    name: creation_time
+                    type: DATETIME
+                    remarks: The creation timestamp for the session id.
+                    constraints: 
+                      nullable: false
+                - column:
+                    name: last_access_time
+                    type: DATETIME
+                    remarks: The last time the session id was used.
+                    constraints:
+                      nullable: false 
+                - column:
+                    name: max_inactive_interval
+                    type: int
+                    remarks: The maximum inactive interval time before a session times out and is invalidated.
+                    constraints: 
+                      nullable: facilities
+                - column:
+                    name: expiry_time
+                    type: DATETIME
+                    remarks: The time this session will expire (last_access_time + max_inactive_interval.)
+                    constraints: 
+                      nullable: false
+                - column:
+                    name: principal_name
+                    type: *string
+                    remarks: The name or UUID of the user for this session.
+        - createIndex:
+            tableName: spring_session
+            indexName: ix__spring_session_expiry_time
+            columns:
+              - column:
+                  name: expiry_time
+        - createIndex:
+            tableName: spring_session
+            indexName: ix__spring_session_principal_name
+            columns:
+              - column:
+                  name: principal_name
+        - createTable:
```

Comment:
Is the shape of this table dictated by Spring? Since these are just key value pairs associated with a session, they could be stored in a JSON or [hstore](https://www.postgresql.org/docs/12/hstore.html) column on the session row itself so that loading a session doesn't involve a `join`.

Reply:
It is unfortunately dictated by Spring; this is the format of the table they expect for postgres. https://github.com/spring-projects/spring-session/blob/main/spring-session-jdbc/src/main/resources/org/springframework/session/jdbc/schema-postgresql.sql